lib/db/lowlevel.go

```diff
@@ -1281,6 +1286,57 @@ func (db *Lowlevel) checkLocalNeed(folder []byte) (int, error) {
 	return repaired, nil
 }
 
+// checkSequencesUnchanged resets delta indexes for any device where the
+// sequence changed.
+func (db *Lowlevel) checkSequencesUnchanged(folder []byte, oldMeta, meta *metadataTracker) error {
+	t, err := db.newReadWriteTransaction()
+	if err != nil {
+		return err
+	}
+	defer t.close()
+
+	var key []byte
+	deleteIndexID := func(devID protocol.DeviceID) error {
+		key, err = db.keyer.GenerateIndexIDKey(key, devID[:], folder)
+		if err != nil {
+			return err
+		}
+		return t.Delete(key)
+	}
+
+	if oldMeta.Sequence(protocol.LocalDeviceID) != meta.Sequence(protocol.LocalDeviceID) {
+		if err := deleteIndexID(protocol.LocalDeviceID); err != nil {
+			return err
+		}
+		l.Infoln("Local sequence changed while repairing - dropping delta indexes")
```

Comment:
Might want to mention the folder in these messages as I guess it might happen for several folders if it happens at all